app/views/tournaments/add_rules.html.erb

```diff
@@ -1,5 +1,6 @@
 <%= javascript_include_tag 'tournaments/add_rules' %>
-
+<h1><a href="<%= events_show_path @tournament.event_id %>"><%=Event.find(@tournament.event_id).name %></a></h1>
```

Comment:
`tournament` belongs to `event`, so I think you can just call `@tournament.event.name` here.